spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/admin/SpannerSchemaUtils.java

```diff
@@ -143,23 +143,50 @@ public Key getKey(Object object) {
 				"CREATE TABLE " + spannerPersistentEntity.tableName() + " ( ");
 
 		StringJoiner columnStrings = new StringJoiner(" , ");
-		spannerPersistentEntity.doWithProperties(
-				(PropertyHandler<SpannerPersistentProperty>) spannerPersistentProperty -> columnStrings
-						.add(getColumnDDLString(
-								spannerPersistentProperty,
-								this.spannerEntityProcessor)));
+
+		addColumnDDLStrings(spannerPersistentEntity, columnStrings);
 
 		stringBuilder.append(columnStrings.toString()).append(" ) PRIMARY KEY ( ");
 
 		StringJoiner keyStrings = new StringJoiner(" , ");
 
+		addPrimaryKeyColumnNames(spannerPersistentEntity, keyStrings);
+
+		stringBuilder.append(keyStrings.toString()).append(" )");
+		return stringBuilder.toString();
+	}
+
+	private <T> void addPrimaryKeyColumnNames(
+			SpannerPersistentEntity<T> spannerPersistentEntity, StringJoiner keyStrings) {
 		for (SpannerPersistentProperty keyProp : spannerPersistentEntity
 				.getPrimaryKeyProperties()) {
-			keyStrings.add(keyProp.getColumnName());
+			if (keyProp.isEmbedded()) {
+				addPrimaryKeyColumnNames(
+						this.mappingContext.getPersistentEntity(keyProp.getType()),
+						keyStrings);
+			}
+			else {
+				keyStrings.add(keyProp.getColumnName());
+			}
 		}
+	}
 
-		stringBuilder.append(keyStrings.toString()).append(" )");
-		return stringBuilder.toString();
+	private <T> void addColumnDDLStrings(
```

Comment:
Generally, we should camel-case acronyms as well, which would make it `addColumnDdlStrings`.

Reply:
Will fix. To be fair nearly all of the functionality we prioritized we haven't received any feedback from the community that would indicate adoption or not :/ 

However, embeddable classes had been discussed internally for a while, and was seen as a somewhat well-known feature from JPA. Furthermore, I think it fits nicely into the plan to support parent-child relationships where the PK components would need to be repeated in parent and child, and where them being embedded would be very convenient.